### ISSUE TITLE
"Implement missing members" assist adds unnecessary unused lifetime parameter<'_>

### ISSUE THREAD

**Issue by Vlad-Shcherbina** [rust-analyzer/rust-analyzer#3986]
rust-analyzer version: 2020-04-13 (c388130)
To reproduce, invoke "Implement missing members" assist on the following program:
```rust
struct S;
impl std::fmt::Display for S {}
```
Expected result:
```rust
impl std::fmt::Display for S {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result { todo!() }
}
```
Actual result:
```rust
impl std::fmt::Display for S {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result { todo!() }
}
```

**Comment by bjorn3** [comment]
When using `#![warn(rust_2018_idioms)]` the result you expected will give a warning, while the actual result doesn't.

**Comment by matklad** [comment]
Yep, this is sort-of "as designed" so far. But I wonder if there's a tracking issue for this lint in rust-lang/rust? Has the idea of restricted this lint to the cases where there isn't an `&` already been floated around?
